deploy/staging/ambuda-runner

```diff
@@ -0,0 +1,159 @@
+#! /usr/bin/bash
+
+AMBUDA_DEPLOY_CONFDIR=${HOME}/deploy-config/
+AMBUDA_WORKSPACE=${HOME}
+
+function usage {
+    echo "Usage: $0 -i <image > <command option> # Setup database"
+    echo "          -c # Setup directory"
+    echo "          -d # Setup database"
+    echo "          -l # Status of Ambuda server"
+    echo "          -s # Start Ambuda"
+    echo "          -x # Stop Ambuda"
+    echo "          -v # Show version"
+    echo "          -h # Help"
+    exit 1
+}
+
+function find_unused_port {
+    BOTTOM_PORT=5001
+    TOP_PORT=5100
+
+    for port in `seq $BOTTOM_PORT $TOP_PORT`
+    do
+        echo "hello" > /dev/tcp/$AMBUDA_HOST_IP/$port
+        if [ "$?" -ne 0 ]
+        then
+            echo "$port"
+            return
+        fi
+    done
+    echo "Error! No unused ports found between $BOTTOM_PORT-$TOP_PORT"
+    exit 1
+}
+
+function init_staging_env {
+    
+    if [ -z "$AMBUDA_STAGING_DIR" ]
+    then
+        exit "Error! Invalid staging directory"
+        exit 1
+    fi
+    
+    mkdir -p "$AMBUDA_STAGING_DIR"
+    if [ "$?" -ne 0 ]
+    then
+        echo "Error! staging directory -- $AMBUDA_STAGING_DIR -- create failed."
+        exit 1
+    else
+        echo "export AMBUDA_STAGING_DIR=$AMBUDA_STAGING_DIR" >> $AMBUDA_STAGING_ENV
+        echo "> Change directory to $AMBUDA_STAGING_DIR"
+    fi
+    
+    
+    echo "export AMBUDA_HOST_IP=$(ip route get 1 | awk '{print $7; exit}')" >> $AMBUDA_STAGING_ENV
+    source $AMBUDA_STAGING_ENV
+
+    echo "export AMBUDA_HOST_PORT=$(find_unused_port)" >> $AMBUDA_STAGING_ENV
+
+    if [ -z "$AMBUDA_IMAGE" ]
+    then
+        echo "Error! Invalid AMBUDA_IMAGE"
+        exit 1
+    else
+        echo "export AMBUDA_IMAGE=$AMBUDA_IMAGE" >> $AMBUDA_STAGING_ENV
+    fi
+}
+
+function setup_staging {
+    if [ ! -e "$AMBUDA_STAGING_ENV" ]
+    then
+        echo "Error! Invalid Ambuda staging environment."
+        exit 1
+    fi
+    source $AMBUDA_STAGING_ENV
+    cd $AMBUDA_STAGING_DIR
+}
+
+function setup_database {
+    docker compose -f ${AMBUDA_DEPLOY_CONFDIR}/docker-compose-dbsetup.yml up
```

Comment:
Why have separate docker-compose files for the database and the rest of it?

Reply:
I created a clean separation between the database and the application:
1. setup database once and almost not touch it. If a need arises, explicitly run. We'll provide hooks to automate it. The database is out of the way for the application developers.
2. in the long run, this helps us to move to the cloud or replace it with a different DB flavor.